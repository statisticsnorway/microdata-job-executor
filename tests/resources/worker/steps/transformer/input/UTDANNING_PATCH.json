{
    "temporalityType": "STATUS",
    "sensitivityLevel": "PERSON_GENERAL",
    "populationDescription": [
        {
            "languageCode": "no",
            "value": "Personer med utdannelse i Norge"
        }
    ],
    "spatialCoverageDescription": [
        {
            "languageCode": "no",
            "value": "Norge"
        }
    ],
    "subjectFields": [
        [
            {
                "languageCode": "no",
                "value": "BEFOLKNING"
            }
        ],
        [
            {
                "languageCode": "no",
                "value": "SAMFUNN"
            }
        ]
    ],
    "dataRevision": {
        "description": [
            {
                "languageCode": "no",
                "value": "Første publisering."
            }
        ],
        "temporalEndOfSeries": false
    },
    "identifierVariables": [
        {
            "shortName": "PERSONID_1",
            "name": [
                {
                    "languageCode": "no",
                    "value": "Personidentifikator"
                },
                {
                    "languageCode": "en",
                    "value": "Person identifier"
                }
            ],
            "description": [
                {
                    "languageCode": "no",
                    "value": "Identifikator for person i microdata"
                },
                {
                    "languageCode": "en",
                    "value": "Identifier for person in microdata"
                }
            ],
            "dataType": "LONG",
            "unitType": {
                "shortName": "PERSON",
                "requiresPseudonymization": true,
                "name": [
                    {
                        "languageCode": "no",
                        "value": "Person"
                    },
                    {
                        "languageCode": "en",
                        "value": "Person"
                    }
                ],
                "description": [
                    {
                        "languageCode": "no",
                        "value": "Person er et enkeltmenneske, individ."
                    },
                    {
                        "languageCode": "en",
                        "value": "Person is a single person, an individual."
                    }
                ]
            },
            "format": "RandomUInt48",
            "valueDomain": {
                "description": [
                    {
                        "languageCode": "no",
                        "value": "Pseudonymisert personnummer"
                    },
                    {
                        "languageCode": "en",
                        "value": "Pseudonymized national identity number"
                    }
                ]
            }
        }
    ],
    "measureVariables": [
        {
            "name": [
                {
                    "languageCode": "no",
                    "value": "Utdanning"
                }
            ],
            "description": [
                {
                    "languageCode": "no",
                    "value": "Utdanningsnivå for person"
                }
            ],
            "subjectFields": [
                [
                    {
                        "languageCode": "no",
                        "value": "Testdata"
                    }
                ]
            ],
            "dataType": "STRING",
            "valueDomain": {
                "codeList": [
                    {
                        "code": "1",
                        "categoryTitle": [
                            {
                                "languageCode": "no",
                                "value": "Grunnskole"
                            }
                        ],
                        "validFrom": "1926-01-01"
                    },
                    {
                        "code": "2",
                        "categoryTitle": [
                            {
                                "languageCode": "no",
                                "value": "Videregående skole"
                            }
                        ],
                        "validFrom": "1926-01-01"
                    },
                    {
                        "code": "3",
                        "categoryTitle": [
                            {
                                "languageCode": "no",
                                "value": "Bachelorgrad"
                            }
                        ],
                        "validFrom": "1926-01-01"
                    },
                    {
                        "code": "4",
                        "categoryTitle": [
                            {
                                "languageCode": "no",
                                "value": "Mastergrad"
                            }
                        ],
                        "validFrom": "1926-01-01"
                    },
                    {
                        "code": "5",
                        "categoryTitle": [
                            {
                                "languageCode": "no",
                                "value": "Doktorgrad"
                            }
                        ],
                        "validFrom": "1926-01-01"
                    }
                ]
            },
            "shortName": "UTDANNING"
        }
    ],
    "attributeVariables": [
        {
            "variableRole": "Start",
            "shortName": "START",
            "dataType": "DATE",
            "valueDomain": {
                "description": [
                    {
                        "languageCode": "no",
                        "value": "Dato oppgitt i dager siden 1970-01-01"
                    }
                ]
            },
            "description": [
                {
                    "languageCode": "no",
                    "value": "Startdato/måletidspunktet for hendelsen"
                },
                {
                    "languageCode": "en",
                    "value": "Startdato/måletidspunktet for hendelsen"
                }
            ],
            "name": [
                {
                    "languageCode": "no",
                    "value": "Startdato"
                },
                {
                    "languageCode": "en",
                    "value": "Start date"
                }
            ]
        },
        {
            "variableRole": "Stop",
            "shortName": "STOP",
            "dataType": "DATE",
            "valueDomain": {
                "description": [
                    {
                        "languageCode": "no",
                        "value": "Dato oppgitt i dager siden 1970-01-01"
                    }
                ]
            },
            "description": [
                {
                    "languageCode": "no",
                    "value": "Startdato/måletidspunktet for hendelsen"
                },
                {
                    "languageCode": "en",
                    "value": "Startdato/måletidspunktet for hendelsen"
                }
            ],
            "name": [
                {
                    "languageCode": "no",
                    "value": "Stoppdato"
                },
                {
                    "languageCode": "en",
                    "value": "Stop date"
                }
            ]
        }
    ],
    "shortName": "UTDANNING"
}