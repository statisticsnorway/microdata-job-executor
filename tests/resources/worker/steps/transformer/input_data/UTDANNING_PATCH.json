{
    "temporalityType": "STATUS",
    "sensitivityLevel": "PERSON_GENERAL",
    "populationDescription": [
        {
            "languageCode": "no",
            "value": "Personer med utdannelse i Norge"
        }
    ],
    "spatialCoverageDescription": [
        {
            "languageCode": "no",
            "value": "Norge"
        }
    ],
    "subjectFields": [
        [
            {
                "languageCode": "no",
                "value": "BEFOLKNING"
            }
        ],
        [
            {
                "languageCode": "no",
                "value": "SAMFUNN"
            }
        ]
    ],
    "dataRevision": {
        "description": [
            {
                "languageCode": "no",
                "value": "Første publisering."
            }
        ],
        "temporalEnd": {
            "description": [
                {
                    "languageCode": "no",
                    "value": "Variabelen blir ikke lenger oppdatert"
                }
            ],
            "successors": [
                "UTDANNING_1",
                "UTDANNING_2"
            ]
        }
    },
    "measureVariables": [
        {
            "name": [
                {
                    "languageCode": "no",
                    "value": "Utdanning"
                }
            ],
            "description": [
                {
                    "languageCode": "no",
                    "value": "Utdanningsnivå for person"
                }
            ],
            "subjectFields": [
                [
                    {
                        "languageCode": "no",
                        "value": "Testdata"
                    }
                ]
            ],
            "dataType": "STRING",
            "valueDomain": {
                "codeList": [
                    {
                        "code": "1",
                        "categoryTitle": [
                            {
                                "languageCode": "no",
                                "value": "Grunnskole"
                            }
                        ],
                        "validFrom": "1926-01-01"
                    },
                    {
                        "code": "2",
                        "categoryTitle": [
                            {
                                "languageCode": "no",
                                "value": "Videregående skole"
                            }
                        ],
                        "validFrom": "1926-01-01"
                    },
                    {
                        "code": "3",
                        "categoryTitle": [
                            {
                                "languageCode": "no",
                                "value": "Bachelorgrad"
                            }
                        ],
                        "validFrom": "1926-01-01"
                    },
                    {
                        "code": "4",
                        "categoryTitle": [
                            {
                                "languageCode": "no",
                                "value": "Mastergrad"
                            }
                        ],
                        "validFrom": "1926-01-01"
                    },
                    {
                        "code": "5",
                        "categoryTitle": [
                            {
                                "languageCode": "no",
                                "value": "Doktorgrad"
                            }
                        ],
                        "validFrom": "1926-01-01"
                    }
                ]
            },
            "shortName": "UTDANNING"
        }
    ],
    "shortName": "UTDANNING"
}