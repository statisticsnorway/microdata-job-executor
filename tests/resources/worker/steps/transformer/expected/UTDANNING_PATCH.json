{"name": "UTDANNING", "temporality": "STATUS", "languageCode": "no", "sensitivityLevel": "PERSON_GENERAL", "populationDescription": "Personer med utdannelse i Norge", "subjectFields": ["BEFOLKNING", "SAMFUNN"], "temporalCoverage": {"start": null}, "measureVariable": {"name": "UTDANNING", "label": "Utdanning", "notPseudonym": true, "dataType": "String", "variableRole": "Measure", "representedVariables": [{"description": "Utdanningsniv\u00e5 for person", "validPeriod": {"start": -16071}, "valueDomain": {"codeList": [{"category": "Grunnskole", "code": "1"}, {"category": "Videreg\u00e5ende skole", "code": "2"}, {"category": "Bachelorgrad", "code": "3"}, {"category": "Mastergrad", "code": "4"}, {"category": "Doktorgrad", "code": "5"}], "missingValues": []}}]}, "identifierVariables": [{"name": "PERSONID_1", "label": "Personidentifikator", "notPseudonym": false, "dataType": "Long", "variableRole": "Identifier", "representedVariables": [{"description": "Identifikator for person i microdata", "validPeriod": {"start": null}, "valueDomain": {"description": "Pseudonymisert personnummer", "unitOfMeasure": "N/A"}}], "format": "RandomUInt48", "keyType": {"name": "PERSON", "label": "Person", "description": "Person er et enkeltmenneske, individ."}}], "attributeVariables": [{"name": "START", "label": "Startdato", "notPseudonym": true, "dataType": "Instant", "variableRole": "Start", "representedVariables": [{"description": "Startdato/m\u00e5letidspunktet for hendelsen", "validPeriod": {"start": null}, "valueDomain": {"description": "Dato oppgitt i dager siden 1970-01-01", "unitOfMeasure": "N/A"}}]}, {"name": "STOP", "label": "Stoppdato", "notPseudonym": true, "dataType": "Instant", "variableRole": "Stop", "representedVariables": [{"description": "Startdato/m\u00e5letidspunktet for hendelsen", "validPeriod": {"start": null}, "valueDomain": {"description": "Dato oppgitt i dager siden 1970-01-01", "unitOfMeasure": "N/A"}}]}]}